File: backend/MediFlow.Api/Models/Doctor.cs
namespace MediFlow.Api.Models;

public class Doctor
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string Bio { get; set; } = string.Empty;
    public string Qualifications { get; set; } = string.Empty;
    public int ExperienceYears { get; set; }
    public decimal ConsultationFee { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Extended profile fields
    public string? ProfilePhoto { get; set; }
    public string? SubSpecialty { get; set; }
    public string? HospitalClinic { get; set; }
    public string? Languages { get; set; }
    public string? Location { get; set; }
    public string? MbbsUniversity { get; set; }
    public string? PhdUniversity { get; set; }
    public string? OtherQualifications { get; set; }
    public string? Certifications { get; set; }
    public int? Age { get; set; }
    public string? RegistrationNumber { get; set; }

    // Navigation properties
    public ICollection<DoctorSpecialty> DoctorSpecialties { get; set; } = new List<DoctorSpecialty>();
    public ICollection<DoctorAvailability> Availabilities { get; set; } = new List<DoctorAvailability>();
    public ICollection<DoctorRating> Ratings { get; set; } = new List<DoctorRating>();
    public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public ICollection<DoctorLeave> Leaves { get; set; } = new List<DoctorLeave>();
}

File: web/mediflow-web/src/types/doctor.ts
export interface DoctorSpecialtyInfo {
  id: number;
  name: string;
}

export interface DoctorReviewDto {
  id: number;
  appointmentId: number;
  patientName: string;
  rating: number;
  review?: string;
  createdAt: string;
}

export interface DoctorDetail {
  id: number;
  userId: number;
  fullName: string;
  qualifications: string;
  bio?: string;
  consultationFee: number;
  experienceYears: number;
  averageRating: number;
  reviewCount: number;
  isActive: boolean;
  specialties: DoctorSpecialtyInfo[];
  profilePhoto?: string;
  subSpecialty?: string;
  hospitalClinic?: string;
  languages?: string;
  location?: string;
  mbbsUniversity?: string;
  phdUniversity?: string;
  otherQualifications?: string;
  certifications?: string;
  age?: number;
  registrationNumber?: string;
  reviews?: DoctorReviewDto[];
}

export interface RankedDoctor extends DoctorDetail {
  rankScore: number;
}

export interface SpecialtyInfo {
  id: number;
  name: string;
  description?: string;
  iconName?: string;
}

export interface DoctorProfileUpdatePayload {
  bio?: string;
  subSpecialty?: string;
  hospitalClinic?: string;
  languages?: string;
  location?: string;
  mbbsUniversity?: string;
  phdUniversity?: string;
  otherQualifications?: string;
  certifications?: string;
  experienceYears?: number;
  age?: number;
  consultationFee?: number;
  profilePhoto?: string;
}
